conda/cli/conda_argparse.py

```diff
@@ -710,8 +710,8 @@ def configure_parser_init(sub_parsers):
     )
     setup_type_group.add_argument(
         "--no-user",
-        action="store_false",
-        help="Don't initialize conda for the current user (default).",
+        action="store_true",
+        help="Don't initialize conda for the current user.",
```

Comment:
It looks like this is a bugfix where we could be using bool(NULL or False) to decide whether to do the action, but the code that does the shell initialization doesn't receive the argument properly.

Reply:
Should we have a "dest=user" here and no default, and only keep args.user

Reply:
That could work.  I'm willing to give it a try.

Reply:
Things get slightly complicated when multiple arguments affect the same `dest`, see:
https://stackoverflow.com/a/14110785

@dholth, let me know if a0d65b355 seems okay, with the caveat that I haven't had time to test it yet myself.